### DISCUSSION TITLE
Major new app version in beta, request feedback

### DISCUSSION THREAD

**Discussion by kebekus** [Akaflieg-Freiburg/enroute#318] [Team Posts]
Dear all,

today I uploaded version 2.22.10 of the app to the beta channel of Google Play.  Despite the inconspicuous version number, this release is a **major** update, and I am expecting some issues and a much longer-than-usual beta phase. If you have the time, could you please try the new version app, and report any issues that you may find? -- Thanks.  

I would also appreciate hearing from you if things work well.

- Does the app feel different? Does it feel responsive? Or perhaps more sluggish than it used to be?
- Is the user interface consistent? Are there glitches in the user interface (dialog windows or menus going off-screen, …)
- Are there graphics problems with the moving map?  

**Background:** the technology on which the app is based (=the Qt library) has seen a major release update.  As support for the old version of Qt is slowly phased out, I have ported the app to the new technology.  If all goes well, I expect that the app will become quite a bit more performant over the next few releases. Additional bonuses: porting to iOS will become much easier, and the new version of Qt simplifies the programmer's job quite a bit.

That said, the port to Qt6 was a major undertaking. The chance that I made a serious mistake at some point is 100%.

**Next steps:** at present, the app compiles and works well on my own devices (and perhaps on yours). The build system works but is not very stable yet.  The parts of the code that needed adjustments have been ported. In Q1 of 2023, I will mostly clean up the code, simplify and stabilize the build system and start to port more code, in order to take advantage of the new library features and of their potential higher performance. I hope that this will result in some noticeable improvements for the user!

Best wishes,

Stefan.

**Comment by think-nice-things** [comment]
Hi Stefan,

thanks for your continued good work.

here's a quick observation with 2.22.10:
the settings page doesn't open. Tapping "settings" simply does nothing.

**Comment by kebekus** [comment]
@think-nice-things Roger. I can reproduce the issue and try to come up with a solution asap.

**Comment by TTC70** [comment]
> 2.22.10 of the app to the beta channel of Google Play. Despite the inconspicuous version number, this release is a major update, and I am expecting some issues and a much lon

Hey Stefan,
Thanks for your hard work!!

Confirming that the issue about "settings" not working, I noticed that there is an issue with double tapping vs keeping pressed:
- double tapping gives correctly the info about that specific point,
- keeping pressed does not do anything,
- double tapping again does not do anything.
If you start with keeping pressed everything works until the next double tap. It seems the issue is related to the "double tap".
If I restart the app, the double tap starts working again, but only for once.

Happy new year to all.

**Comment by kebekus** [comment]
@TTC7 Roger. I can reproduce the issue. I am amazed and I will try and fix the problem.

**Comment by kebekus** [comment]
Dear all, the problem with the "Settings" page should be fixed in version 2.22.11, which is in beta now. I will next look at the tap-and-hold problem reported by @TTC70 

Thank you all for your speedy reports! I would never be able to find all these issues without help.

Best,

Stefan.

**Comment by kebekus** [comment]
Dear all, I just uploaded version 2.22.12, which hopefully solved the issue reported by @TTC70 .

Thanks again!

**Comment by TTC70** [comment]
Stefan,
a deep dive into this issue after downloading the last version 2.22.12.
I turn the app on, the double-tap and tap+hold (in this order) work.
If I ONLY tap+hold, it always works. Until I double-tap. 
Starting from the first double-tap neither tap+hold nor double-tap work.
The additional info is: if I choose almost any voices of the main menù (I have to actually open one of the menu choices, I cannot just show the menu itself) both double-tap AND tap+hold work... but with the conditions explained up here (that is, AFTER the double-tap both of them stop working, until the next menu choice).
Just to recap:
- double-tap works only once. It breaks tap+hold. It starts working again after the user shows any manu page;
- tap+hold always works unless double-tap is used.

I hope this helps.

P.S. "Settings" menu works.

**Comment by kebekus** [comment]
@TTC70 Ooops. It seems that we ran into a bug in Qt6! I implemented a workaround, and uploaded v2.22.13 to Google Play/Beta. Can you please check that double-tap and tap-and-hold now works as expected?

**Comment by TTC70** [comment]
Stefan,
At the moment the functionality of the app is at its best.
Also double tap works properly.
Thank you for your patience.


Il Dom 8 Gen 2023, 08:55 Stefan Kebekus ***@***.***> ha
scritto:

**Comment by kebekus** [comment]
@TTC70 Glad to hear that. Thank you for your help!
